sys var: add `tidb_enable_shared_lock_upgrade` (#18447)
pingcap/docs: system-variables.md

- Value options: `OFF`, `ON`
- This variable controls whether TiDB enables chunk objects cache. If the value is `ON`, TiDB prefers to use the cached chunk object and only requests from the system if the requested object is not in the cache. If the value is `OFF`, TiDB requests chunk objects from the system directly.

> ### tidb_enable_shared_lock_upgrade <span class="version-mark">New in v8.3.0</span>

Author: ```suggestion
### tidb_enable_shared_lock_promotion <span class="version-mark">New in v8.3.0</span>
```

> ### `tidb_enable_shared_lock_upgrade` <span class="version-mark">New in v8.3.0</span>

Author: ```suggestion
### tidb_enable_shared_lock_upgrade <span class="version-mark">New in v8.3.0</span>
```

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Boolean
- Default value: `OFF`

> - This variable controls whether to enable the function of upgrading shared locks to exclusive locks. When the variable value is `ON`, TiDB will try to upgrade shared locks to exclusive locks to reduce the probability of deadlock. The default value of this variable is `OFF`, which means that the function of upgrading shared locks to exclusive locks is disabled.

Reviewer: ```suggestion
- This variable controls whether to enable the function of upgrading shared locks to exclusive locks. When the variable value is `ON`, TiDB will try to upgrade shared locks to exclusive locks. The default value of this variable is `OFF`, which means that the function of upgrading shared locks to exclusive locks is disabled.
```

> - This variable controls whether to enable the function of upgrading shared locks to exclusive locks. When the variable value is `ON`, TiDB will try to upgrade shared locks to exclusive locks to reduce the probability of deadlock. The default value of this variable is `OFF`, which means that the function of upgrading shared locks to exclusive locks is disabled.

Author: ```suggestion
- This variable controls whether to enable the function of upgrading shared locks to exclusive locks. When the variable value is `ON`, TiDB will try to upgrade the shared lock to an exclusive lock， and add a pessimistic lock for the `for share` statement. The default value of this variable is `OFF`, which means that the function of upgrading shared locks to exclusive locks is disabled.
```

> - If this variable and [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-new-in-v40) are enabled at the same time, the setting of `tidb_enable_noop_functions` will be overwritten.

Reviewer 2: ```suggestion
- If this variable and [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-new-in-v40) are enabled at the same time, this variable will take effect for the the `SELECT LOCK IN SHARE MODE` statement.
```

> - If this variable and [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-new-in-v40) are enabled at the same time, the setting of `tidb_enable_noop_functions` will be overwritten.

Author: ```suggestion
- If this variable and [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-new-in-v40) are enabled at the same time, for the `SELECT LOCK IN SHARE MODE` statement, TiDB will enable this function first.
```

> - Enabling this variable takes effect on the `SELECT LOCK IN SHARE MODE` statement whether [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-new-in-v40) is enabled or not.

Author: ```suggestion
- Enabling this variable takes effect on the `SELECT LOCK IN SHARE MODE` statement, regardless of whether [`tidb_enable_noop_functions`](/system-variables.md#tidb_enable_noop_functions-new-in-v40) is enabled or not.
```

> - If this variable and [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-new-in-v40) are enabled at the same time, for the `SELECT LOCK IN SHARE MODE` statement, TiDB will enable this function first.

Author: ```suggestion
- If this variable and [`tidb_enable_noop_functions`](#tidb_enable_noop_functions-new-in-v40) are enabled at the same time, this variable will take effect for the the `SELECT LOCK IN SHARE MODE` statement.
```
